### PR TITLE
Fix safeToBetaReduce in UPLC inliner and update uplc test.

### PR DESCRIPTION
**Pull request by thealmarty**
Separating this out to see the changes in tests more easily.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Changelog fragments have been written (if appropriate)
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting master unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

File: plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden [hunk 1]
```diff
@@ -2,7 +2,7 @@
   (lam
     a_5
     [
-      [ f_1 [ [ a_5 (con integer 0) ] (con integer 1) ] ]
+      [ f_1 [ [ g_2 (con integer 1) ] (con integer 0) ] ]
```
Comment (michaelpj) [inline plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden:5]: okay, now _this_ looks busted. That's a free variable where we didn't have one before.

Comment (michaelpj) [reply plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden:5]: Oh wait, `g` is free in the input! and a is bound to g. So this is actually fine, we've just inlined a, which seems okay

**Comment by thealmarty** [comment]
Unfortunately the script context sizes increased. :( Also, in `no-fib-test` and `script-contexts-test` some of the `SizeTest` failed. :(

**Comment by michaelpj** [comment]
Well, it's going to change how much we inline, so I guess it's a toss up between getting better and worse!

File: plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden (hunk 1 above)
Comment (thealmarty) [reply plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden:5]: I'm not sure I understand. These two outputs are semantically different, I think. Also I don't know why we say *inline* `a` here, I think this is beta reduction?

Comment (michaelpj) [reply plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden:5]: Sorry, a is not bound to g. a is bound to `\x y -> g y x`. This is UPLC, so it's an immediately applied lambda, see `a` at line 3 and the definition on line 9. We've indeed _inlined_ one occurrence of `a`, replacing it with `\x y -> g y x`, and then reducing that lambda. We haven't beta reduced the original application - it's still there!

Comment (thealmarty) [reply plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden:5]: I see that now. Thanks!

Comment (thealmarty) [reply plutus-core/untyped-plutus-core/test/Transform/callsiteInline.uplc.golden:5]: I guess we just didn't notice that `a` wasn't inlined before! Great news that this is more right now.

**Review by michaelpj — APPROVED** [review]

**Pull request merged**